src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -3184,6 +3242,33 @@ set_arg_at_index(int index, const xrt::bo& glb)
 }
 
 }
+
+////////////////////////////////////////////////////////////////
+// xrt::run::command_error
+////////////////////////////////////////////////////////////////
+namespace xrt {
+
+run::command_error::
+command_error(ert_cmd_state state, const std::string& msg)
+  : m_impl(std::make_shared<run::command_error_impl>(state, msg))
+{}
+
+ert_cmd_state
+run::command_error::
+get_command_state() const
+{
+  return m_impl->m_state;
+}
+
+const char*
+run::command_error::
+what() const noexcept
+{
+  return m_impl->m_message.c_str();
+}
```

Comment:
Shouldn't we also include string representation of the `m_state`? I just looked up the following examples:

1. fs exceptions report reasons for the failure as well: https://en.cppreference.com/w/cpp/filesystem/copy_file
2. system error translates the meaning of error code: https://en.cppreference.com/w/cpp/error/system_error

Reply:
Yeah, it's just painful to translate the error codes, we have no such map yet.

Reply:
Added string representation at throw time.